test/test-integration.js

```diff
@@ -86,11 +94,39 @@ marionette('gaia-switch', function() {
   test('gaia-switch after user action', function() {
     ['click', 'tap'].forEach(function(action) {
       switches.forEach(function(aSwitch) {
+        if (!aSwitch.checkedOnAction) {
+          return;
+        }
         aSwitch.checkedOnAction.forEach(function(checked) {
-          aSwitch.element[action]();
-          assert.equal(isChecked(aSwitch.element), checked, aSwitch.selector);
+          if (aSwitch.enabled) {
+            aSwitch.element[action]();
+            assert.equal(isChecked(aSwitch.element), checked, aSwitch.selector);
+          } else {
+            try {
+              aSwitch.element[action]();
+            } catch (err) {
+              // NOTE: selenium checks for isEnabled does not support custom
+              // elements and will always return true. Although from a11y
+              // standpoint it is disabled.
+              // See: https://code.google.com/p/selenium/source/browse/
+              //              javascript/atoms/dom.js#324
+              assert.equal(err.type, 'ElementNotAccessibleError');
+            }
+          }
         });
       });
     });
   });
+
+  suite('invalid gaia-switches', function() {
+    test('unlabelled gaia-switch', function() {
+      try {
+        switches[4].element.click();
+      } catch (err) {
```

Comment:
Checks run always. By default they only get logged in gecko.log (marionette default log). However if the flag (technically called marionette capability) is set to true like this: https://github.com/yzen/gaia-switch/blob/bug-1204631/test/test-integration.js#L31-L33 an error is going to be thrown instead of a log message.